tests/components/homematicip_cloud/test_config_flow.py

```diff
@@ -76,92 +108,104 @@
     assert result["type"] == "form"
+    assert result["step_id"] == "link"
+    assert result["errors"] == {"base": "press_the_button"}
 
 
-async def test_init_flow_user_show_form(hass):
+async def test_init_flow_show_form(hass):
     """Test config flow shows up with a form."""
-    flow = config_flow.HomematicipCloudFlowHandler()
-    flow.hass = hass
 
-    result = await flow.async_step_user(user_input=None)
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": "user"}
+    )
     assert result["type"] == "form"
+    assert result["step_id"] == "init"
 
 
 async def test_init_already_configured(hass):
     """Test accesspoint is already configured."""
-    MockConfigEntry(
-        domain=const.DOMAIN, data={const.HMIPC_HAPID: "ABC123"}
-    ).add_to_hass(hass)
+    MockConfigEntry(domain=const.DOMAIN, unique_id="ABC123").add_to_hass(hass)
     config = {
         const.HMIPC_HAPID: "ABC123",
         const.HMIPC_PIN: "123",
         const.HMIPC_NAME: "hmip",
     }
 
-    flow = config_flow.HomematicipCloudFlowHandler()
-    flow.hass = hass
+    with patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_checkbutton",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            const.DOMAIN, context={"source": "user"}, data=config
+        )
 
-    result = await flow.async_step_init(user_input=config)
     assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
 
 
 async def test_import_config(hass):
     """Test importing a host with an existing config file."""
-    flow = config_flow.HomematicipCloudFlowHandler()
-    flow.hass = hass
-
-    result = await flow.async_step_import(
-        {
-            hmipc.HMIPC_HAPID: "ABC123",
-            hmipc.HMIPC_AUTHTOKEN: "123",
-            hmipc.HMIPC_NAME: "hmip",
-        }
-    )
+    config = {
+        const.HMIPC_HAPID: "ABC123",
+        const.HMIPC_AUTHTOKEN: "123",
+        const.HMIPC_NAME: "hmip",
+    }
+
+    with patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_checkbutton",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_setup",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_register",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            const.DOMAIN, context={"source": "import"}, data=config
+        )
 
     assert result["type"] == "create_entry"
     assert result["title"] == "ABC123"
-    assert result["data"] == {
-        hmipc.HMIPC_HAPID: "ABC123",
-        hmipc.HMIPC_AUTHTOKEN: "123",
-        hmipc.HMIPC_NAME: "hmip",
-    }
+    assert result["data"] == {"authtoken": "123", "hapid": "ABC123", "name": "hmip"}
+    assert result["result"].unique_id == "ABC123"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4887152)